ln-simln-jamming/src/attack_interceptor.rs

```diff
@@ -116,8 +117,7 @@ mod tests {
             fn setup_for_network(&self) -> Result<crate::attacks::NetworkSetup, BoxError>;
             async fn intercept_attacker_htlc(&self, req: InterceptRequest) -> Result<Result<CustomRecords, ForwardingError>, BoxError>;
             async fn intercept_attacker_receive(&self,_req: InterceptRequest) -> Result<Result<CustomRecords, ForwardingError>, BoxError>;
-            async fn run_custom_actions(&self, attacker_nodes: HashMap<String, Arc<tokio::sync::Mutex<SimNode<SimGraph>>>>, simulation_completed_check: tokio::sync::oneshot::Sender<()>, shutdown_listener: Listener) -> Result<(), BoxError>;
-            async fn simulation_completed(&self, _start_reputation: NetworkReputation) -> Result<bool, BoxError>;
+            async fn run_attack(&self, _start_reputation: NetworkReputation, attacker_nodes: HashMap<String, Arc<tokio::sync::Mutex<SimNode<SimGraph>>>>, shutdown_listener: Listener) -> Result<(), BoxError>;
```

Comment:
nitnit: if we'd done this refactor before the core change, would be able to have slightly less churn here (wouldn't need to add and remove the oneshot).

More a general structural comment than something that needs to change here!